CMP1903_A1_2324/Die.cs

```diff
@@ -2,22 +2,19 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 
-namespace CMP1903_A1_2324
-{
-    internal class Die
-    {
-        /*
-         * The Die class should contain one property to hold the current die value,
-         * and one method that rolls the die, returns and integer and takes no parameters.
-         */
-
-        //Property
-
-
-        //Method
-
-
+namespace CMP1903_A1_2324 {
+  internal class Die {
+    private int _dieValue; //Private attribute
+    public int DieValue { 
+        get => _dieValue; set => _dieValue = value; //Getter and setter methods for encapsulation
+    }
+    public int Roll() {
+        var rand = new Random(); //create new random object
+        Thread.Sleep(100); //Pause for a moment to ensure new random numbers are generated
+        return rand.Next(1,7); //return an integer from between 1 and 6
```

Comment:
All requirements have been met for the Die class from what i can see!

Great formatting towards the style guide here with a good use of attributes

Die class is very readable with great comments 

Error of Random generating the same value based on seed has been handled correctly